assets/gpt-4-turbo_config.yml

```diff
@@ -0,0 +1,43 @@
+api_keys:
+  gpt-3.5-turbo-1106:
+    - api_key: sk-xxxxx
+      organization: org-xxxxxx
+      model: gpt-3.5-turbo-1106
+  gpt-4-1106-preview:
+    - api_key: sk-xxxxx
+      organization: org-xxxxxx
+      model: gpt-4-1106-preview
+
+
+default_request_type: openai # or xagent
+default_completion_kwargs:
+  model: gpt-4-1106-preview
+  temperature: 0.2
+  request_timeout: 60
+
+enable_summary: true
+summary:
+  single_action_max_length: 2048
+  max_return_length: 12384
```

Comment:
max_return_length is the model's output limit? How do we take advantage of the 128k input?

Reply:
If you want longer context for XAgent input, just set `summary.max_return_length` and other length limit to anything you want.

Reply:
@Umpire2018 suggests `max_return_length` of 4096 and @luyaxi that it's input so 128k - 4k = 124k. Which should we use?

Reply:
You can verify my advice by checking the summarize.py in XAgent/agent. By the way, you should not set the value too high because there will be extra cost in running XAgent, so make sure `summary.max_return_length`+`max_plan_length` < 128k - extra_cost - 4096. The extra cost is around 1000 tokens.